subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/CrossProjectConfigurationReportingGradle.kt

```diff
@@ -176,8 +176,8 @@ class CrossProjectConfigurationReportingGradle private constructor(
     private
     fun Action<in Project>.withCrossProjectModelAccessCheck(): Action<Project> {
         val originalAction = this@withCrossProjectModelAccessCheck
-        return Action<Project> {
-            val originalProject = this@Action
+        return Action<Project> inner@ {
+            val originalProject = this@inner
```

Comment:
The error here is a little confusing, but basically there's the outer Action-scope for the extension and the inner Action-scope.  I'm not sure which Action scope this was intended to refer to.